lib/src/Uncertainty/Algorithm/Sensitivity/RankSobolSensitivityAlgorithm.cxx

```diff
@@ -0,0 +1,432 @@
+      data(0, 1) = firstOrderConfidenceInterval.getLowerBound()[k];
+      data(1, 0) = (k + 1);
+      data(1, 1) = firstOrderConfidenceInterval.getUpperBound()[k];
+      graph.add(Curve(data, "red", "solid", 2, ""));
+    }
+  }
+  // Set bounding box
+  Point lowerBound(2, -0.1);
+  lowerBound[0] = 1.0 - dimension / 10.0;
+  Point upperBound(2, 1.1);
+  const Scalar descriptionMargin = 1.6 * (dimension - 1.0) / (dimension + 2.0);
+  upperBound[0] = dimension + descriptionMargin;
+  graph.setBoundingBox(Interval(lowerBound, upperBound));
+  graph.setIntegerXTick(true);
+  graph.setLegendPosition("topright");
+  return graph;
+}
+
+/** Method that draw (plot) the sensitivity graph */
+Graph RankSobolSensitivityAlgorithm::draw() const
+{
+  Graph graph(DrawSobolFirstIndices(inputDescription_, getAggregatedFirstOrderIndices()));
+  if (outputDesign_.getDimension() > 1)
+    graph.setTitle(OSS() << "Aggregated Sobol' indices - " << getClassName());
+  else
+    graph.setTitle(OSS() << "Sobol' indices - " << getClassName());
+  const UnsignedInteger dimension = aggregatedFirstOrderIndices_.getDimension();
+
+  // Draw confidence intervals
+  if (confidenceLevel_ > 0.0)
+  {
+    const Interval foInterval(getFirstOrderIndicesInterval());
+
+    // transform data
+    Sample data(2, 2);
+    for (UnsignedInteger k = 0; k < dimension; ++k)
+    {
+      // Relative to FirstOrder
+      data(0, 0) = (k + 1);
+      data(0, 1) = foInterval.getLowerBound()[k];
+      data(1, 0) = (k + 1);
+      data(1, 1) = foInterval.getUpperBound()[k];
+      graph.add(Curve(data, "red", "solid", 2, ""));
+    }
+  }
+  return graph;
+}
+
+
+
+// Function that computes aggregated indices  using Vi + variance
+// Indices are aggregated with respect to output dimensions
+Point RankSobolSensitivityAlgorithm::computeAggregatedIndices(const Sample & Vi,
+    const Point & variance) const
+{
+  // Generic implementation
+  const UnsignedInteger inputDimension = Vi.getDimension();
+  const UnsignedInteger outputDimension = Vi.getSize();
+  if (inputDimension == 1)
+  {
+    return Point(Vi[0]);
+  }
+
+  // Compute sum of Var(Y^k)
+  Scalar sumVariance = variance.norm1();
+
+  // Compute aggregated indices
+  return Point(Vi.computeMean() * (outputDimension / sumVariance));
+}
+
+
+
+/** void method that computes confidence interval */
+void RankSobolSensitivityAlgorithm::computeBootstrapDistribution() const
+{
+  // Build interval using sample variance
+  // Mean reference is the Sensitivity values
+  const Point FirstOrder(getFirstOrderIndices());
+
+  if (bootstrapSize_ > 0)
+  {
+    // Temporary samples that stores the first/total indices
+    const UnsignedInteger inputDimension = inputDescription_.getSize();
+    Sample bsFO(0, inputDimension);
+
+    const UnsignedInteger size = size_;
+    
+    UnsignedInteger bootstrapSampleSize(ResourceMap::GetAsScalar("RankSobolSensitivityAlgorithm-DefaultBootstrapSampleRatio")*inputDesign_.getSize());
```

Comment:
Does this work if ratio*inputsize is not an integer? i.e., does it truncate it to cast into a UnsignedInteger ?

Reply:
it returns the nearest integer (floor)